sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -72,7 +72,8 @@ case class HadoopFsRelation(
     val compressionFactor = sqlContext.conf.fileCompressionFactor
     val defaultSize = (location.sizeInBytes * compressionFactor).toLong
     location match {
-      case cfi: CatalogFileIndex if sparkSession.sessionState.conf.fallBackToHdfsForStatsEnabled =>
+      case cfi: CatalogFileIndex if sparkSession.sessionState.conf.fallBackToHdfsForStatsEnabled &&
+          location.sizeInBytes == sqlContext.conf.defaultSizeInBytes =>
```

Comment:
The point @maropu and I were making is, `location.sizeInBytes == sqlContext.conf.defaultSizeInBytes` doesn't mean the table stats are not available. `sqlContext.conf.defaultSizeInBytes` is configurable, and it's possible that the table stats are the same as `sqlContext.conf.defaultSizeInBytes`, in which case we shouldn't fallback to HDFS.

Reply:
Yes, I agree. my point was, if the table statistics is not empty, it will not fallback to hdfs even without the condition. So, the PR itself isn't necessary. I will close this to #24715 
Thanks @dongjoon-hyun , @cloud-fan @maropu for the feedbacks